src/main/java/com/google/api/generator/gapic/composer/rest/HttpJsonServiceStubClassComposer.java

```diff
@@ -0,0 +1,508 @@
+            ConcreteReference.builder()
+                .setClazz(List.class)
+                .setGenerics(TypeNode.STRING.reference())
+                .build());
+
+    extractorVarType =
+        TypeNode.withReference(
+            ConcreteReference.builder()
+                .setClazz(Map.class)
+                .setGenerics(TypeNode.STRING.reference(), fieldsVarGenericType.reference())
+                .build());
+
+    expr =
+        methodMaker
+            .apply(
+                "setQueryParamsExtractor",
+                Arrays.asList(
+                    createFieldsExtractorAnonClass(
+                        protoMethod,
+                        extractorVarType,
+                        protoMethod.httpBindings().queryParameters(),
+                        "putQueryParam")))
+            .apply(expr);
+
+    extractorVarType = TypeNode.STRING;
+    expr =
+        methodMaker
+            .apply(
+                "setRequestBodyExtractor",
+                Arrays.asList(
+                    createFieldsExtractorAnonClass(
+                        protoMethod,
+                        extractorVarType,
+                        protoMethod.httpBindings().bodyParameters(),
+                        "toBody")))
+            .apply(expr);
+    expr = methodMaker.apply("build", Collections.emptyList()).apply(expr);
+
+    return Collections.singletonList(expr);
+  }
+
+  private List<Expr> setResponseParserExpr(Method protoMethod) {
+    BiFunction<String, List<Expr>, Function<MethodInvocationExpr, MethodInvocationExpr>>
+        methodMaker = getMethodMaker();
+
+    MethodInvocationExpr expr =
+        MethodInvocationExpr.builder()
+            .setStaticReferenceType(
+                FIXED_REST_TYPESTORE.get(ProtoMessageResponseParser.class.getSimpleName()))
+            .setMethodName("newBuilder")
+            .setGenerics(Collections.singletonList(protoMethod.outputType().reference()))
+            // .setArguments(Arrays.asList(m))
+            .build();
+
+    expr =
+        methodMaker
+            .apply(
+                "setDefaultInstance",
+                Arrays.asList(
+                    MethodInvocationExpr.builder()
+                        .setStaticReferenceType(protoMethod.outputType())
+                        .setMethodName("getDefaultInstance")
+                        .setReturnType(protoMethod.outputType())
+                        .build()))
+            .apply(expr);
+    expr = methodMaker.apply("build", Collections.emptyList()).apply(expr);
+
+    return Collections.singletonList(expr);
+  }
+
+  private Expr createFieldsExtractorAnonClass(
+      Method method,
+      TypeNode extractorReturnType,
+      Set<String> httpBindingFieldNames,
+      String serializerMethodName) {
+    List<Expr> bodyExprs = new ArrayList<>();
+
+    Expr returnExpr = null;
+    VariableExpr fieldsVarExpr = null;
+    Expr serializerExpr = null;
+    if (!extractorReturnType.isProtoPrimitiveType()) {
```

Comment:
Could we please flip the if-blocks so the shorter one comes first?

Reply:
Is there a best practice about this? conditional `return/continue` is preferred to go first but it is not quite the same as if-else block, after which method continues. Well, anyways, flipped.